### PR TITLE
initial add of K8s automation

### PR DESCRIPTION
**Pull request by metadave**

### CODE REVIEW

**Comment by metadave** [comment]
Fixed a few issues with secrets and deployment, I'll need to pair with a SUMO dev to get this running completely. 

```
$ kubectl -n sumo-dev-metadave get pods
NAME                            READY     STATUS             RESTARTS   AGE
sumo-dev-web-4144724747-66z75   1/1       Running            0          9m
sumo-dev-web-4144724747-n2pvz   0/1       CrashLoopBackOff   1          4m
sumo-dev-web-4144724747-q9brx   1/1       Running            0          9m
```

```
# Pastebin OSG25bPi
[2018-02-05 19:47:47 +0000] [1] [INFO] Starting gunicorn 19.4.5
[2018-02-05 19:47:47 +0000] [1] [INFO] Listening at: http://0.0.0.0:8000 (1)
[2018-02-05 19:47:47 +0000] [1] [INFO] Using worker: meinheld.gmeinheld.MeinheldWorker

Pods in running state show this in the logs:

...
[2018-02-05 19:49:51 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:38)
[2018-02-05 19:49:51 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:39)
[2018-02-05 11:49:51 +0000] [39] [INFO] Worker exiting (pid: 39)
[2018-02-05 11:49:51 +0000] [38] [INFO] Worker exiting (pid: 38)
[2018-02-05 19:49:52 +0000] [46] [INFO] Booting worker with pid: 46
[2018-02-05 19:49:52 +0000] [47] [INFO] Booting worker with pid: 47
Raven is not configured (logging is disabled). Please see the documentation for more information.
Raven is not configured (logging is disabled). Please see the documentation for more information.
[2018-02-05 19:50:22 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:46)
[2018-02-05 19:50:22 +0000] [1] [CRITICAL] WORKER TIMEOUT (pid:47)
[2018-02-05 11:50:22 +0000] [47] [INFO] Worker exiting (pid: 47)
[2018-02-05 11:50:22 +0000] [46] [INFO] Worker exiting (pid: 46)
[2018-02-05 19:50:23 +0000] [54] [INFO] Booting worker with pid: 54
[2018-02-05 19:50:23 +0000] [55] [INFO] Booting worker with pid: 55
```

While the pod in error state shows this:

```
kubectl -n sumo-dev-metadave logs sumo-dev-web-4144724747-7pmjr
./bin/run-prod.sh: 7: exec: gunicorn: not found
```

File: k8s/regions/oregon-b/dev.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+SUMO_ENV_HTTPS: True
+SUMO_ENV_MEDIA_URL: https://dev-cdn.sumo.mozilla.net/
+SUMO_ENV_PIPELINE_ENABLED: True
+SUMO_ENV_PORT: 8000
+SUMO_ENV_READ_ONLY: True
+SUMO_ENV_SESSION_COOKIE_SECURE: True
+SUMO_ENV_SITE_URL: http://sumo-dev.frankfurt.moz.works
+SUMO_ENV_STAGE: False
+SUMO_IMAGE_PULL_POLICY: IfNotPresent
+SUMO_IMAGE_TAG: full-latest
```
Comment (jgmize) [inline k8s/regions/oregon-b/dev.yaml]: using a mutable tag in combination with the IfNotPresent pull policy is likely resulting in an older, broken image being used on one of the nodes, causing the `./bin/run-prod.sh: 7: exec: gunicorn: not found` error you described in https://github.com/mozilla/kitsune/pull/3034#issuecomment-363203330

**Comment by jgmize** [comment]
findings so far on the nginx worker timeouts:

- the version deployed by Deis Workflow does not appear to be having the issue
- the only significant difference I've found between the environment variables set in the sumo-dev-metadave web containers @metadave has deployed and the sumo-dev containers deployed by Deis Workflow  is READ_ONLY=True in the former and False in the latter.
  - I tried setting READ_ONLY=False via kubectl edit but still got the same nginx worker timeouts.

**Comment by metadave** [comment]
I pushed a few more updates:
- bumping up the CPU limit solved the worker timeouts, we'll have to carefully tweak these values for prod 🎉 
- added additional invoke targets for celery/cron deployments
- celery runs, I'm not sure how to test it (yet)
- cron deployment has not been tested, I didn't want to run it without running it by @glogiotatidis first

**Comment by glogiotatidis** [comment]
I added new-relic-license-key in sumo secrets yaml. 
 
@metadave can you please update to include 
 - `NEW_RELIC_LICENSE_KEY`
 - `NEW_RELIC_APP_NAME=sumo-dev-frankfurt`

**Comment by metadave** [comment]
@glogiotatidis updated. I also pushed an update to the private repo to add a new `frankfurt` subdir to help organize secrets.

**Comment by metadave** [comment]
merging as-is, the team will iterate over this in the next few months.

**Pull request merged**